client/src/app/users/user.component.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { ComponentFixture, TestBed, async } from "@angular/core/testing";
+import { User } from "./user";
+import { UserComponent } from "./user.component";
+import { UserListService } from "./user-list.service";
+import { Observable } from "rxjs";
+import { PipeModule } from "../../pipe.module";
+
+describe("User component", () => {
+
+    let userComponent: UserComponent;
+    let fixture: ComponentFixture<UserComponent>;
+
+    let userListServiceStub: {
```

Comment:
Good question!

Because JS is dynamically typed and because we didn't specify an interface/type for the service, we only need to provide the functions that are called in this test. I *think* that if we had provided an interface-y thing for the service, then our stub would probably need to implement that interface, although we could provide empty implementations of functions we never call.